src/Particle/ParticleSet.cpp

```diff
@@ -659,8 +686,14 @@ void ParticleSet::acceptMove(Index_t iat, bool partial_table_update)
 
 void ParticleSet::flex_donePbyP(const RefVector<ParticleSet>& P_list)
 {
-  for (int iw = 0; iw < P_list.size(); iw++)
-    P_list[iw].get().donePbyP();
+  if (P_list.size() > 1)
+  {
+#pragma omp parallel for
```

Comment:
Currently, donePbyP is CPU only. The SK computation is not light and needs threading over walkers. So implementing loop and omp in place.

Reply:
This is still consistent with our current crowd design. The parallel for is only active with 1 crowd or OpenMP nested threading enabled.

Reply:
So this should only be used in the case of CPU threads are a multiple of the total walkers i.e. 64 with 8 walkers.
Then there would be 8 crowds and 8 top level threads.  But then P_List size would never be > 1
So...
I don't get it.  It just seems to be a way to go.
1 crowd many walkers when it should really still be parallel over crowds.

now I have an omp loop that works just like the legacy code.  This implicitly requires a full population sync and redistribution of what is local to each thread. That seems simple because openmp hiding it but its just handing out pieces of the P_list which of course hold all sorts of pointers to all sorts of things which can later on cause race conditions.  

If the reason to have it is to write an openmp offload kernel in the future it seems like a kernel incompatible with batching that isn't heavily synchronized.